### PR TITLE
0.9 to 0.10 Metadata Migration Script

### PR DESCRIPTION
**Pull request by ulixius9**
### Describe your changes :

0.9 to 0.10 Metadata Migration Script

#
### Type of change :

- [x] Bug fix
- [x] Improvement
#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/metadata.py [hunk 1]
```diff
@@ -334,6 +352,203 @@ def fetch_glossary_terms(self) -> GlossaryTerm:
...
+
+    def fetch_charts(self) -> Chart:
+        """fetch charts method
+
+        Returns:
+            Chart:
+        """
+        after = None
+        while True:
+            chart_entities = self.metadata.list_entities(
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/metadata.py:385]: This logic seems that could be reused with a function accepting `entity: Type[T], fields: List[str]` 🤔

File: ingestion/src/metadata/ingestion/source/migrate.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+class Policy:
+
+    policy_dict: dict
+
+    def __init__(self, policy_dict) -> None:
+        self.policy_dict = policy_dict
+
+
+class Tag:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/migrate.py:37]: Mixed feelings when shadowing JSON Schema classes

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/migrate.py:37]: Also, why do we have a `migrate` source? We already have an `OpenMetadata` source we could use

File: ingestion/src/metadata/ingestion/stage/migrate.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            OpenMetadataConnection.parse_obj(self.metadata_config)
+        )
+
+    @classmethod
+    def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
+        config = FileSinkConfig.parse_obj(config_dict)
+        return cls(config, metadata_config)
+
+    def stage_record(self, record: Entity) -> None:
+        if isinstance(record, Table):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/stage/migrate.py:117]: This can indeed be a singledispatch

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/stage/migrate.py:117]: We can use this `stage` as a chance to try out a new structure and clean the rest of the ingestion later on

File: ingestion/src/metadata/ingestion/stage/migrate.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        if self.wrote_glossary_term:
+            self.glossary_term_file.write(",\n")
+        self.wrote_glossary_term = True
+        self.glossary_term_file.write(record.json())
+
+    def write_database_service(self, record: Entity) -> None:
+        if self.wrote_database_service:
+            self.database_service_file.write(",\n")
+        self.wrote_database_service = True
+        tag_obj = json.dumps(record.database_service_dict)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/stage/migrate.py:189]: Why not .json()?

File: ingestion/src/metadata/ingestion/stage/migrate.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        )
+        self.role_file = open(f"{self.config.filename}/role.json", "w")
+        self.policy_file = open(f"{self.config.filename}/policy.json", "w")
+        self.pipeline_service_file = open(
+            f"{self.config.filename}/pipeline_service.json", "w"
+        )
+        self.messaging_service_file = open(
+            f"{self.config.filename}/messaging_service.json", "w"
+        )
+        self.table_file.write("[\n")
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/stage/migrate.py:77]: we can kill this if we then read the file line by line, each line being a JSON.

```
with open(filename) as file:
    for line in file:
        # push to OM
```

**Comment by ulixius9** [comment]
@pmbrull, for services I'm only migrating the description and owner details.

**Review by harshach — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Pull request merged**